### PR TITLE
ModularTorchModel load_from_pretrained function fix

### PR DESCRIPTION
**Pull request by Harindhar10**
In the load_from_pretrained function of ModularTorchModel, when a checkpoint file is provided as the source of pretrained weights, the target model’s restore function is invoked. As a result, the optimizer state and global step from the source model are also loaded. This appears to be a bug because:

- The restore function was designed for resuming training on the same model, not for transferring weights to a different one.
- Loading the optimizer state (which contains accumulated gradients from the pretraining dataset) and the global step (which shortens early optimizer warmup steps) is inappropriate when fine-tuning on a new dataset and task.
- This issue does not occur when weights are loaded from a model class rather than from a checkpoint file.
 
Code snippet to replicate the bug:
https://colab.research.google.com/drive/1auXGnn6PkwqT8aHTiPMxKJC0uqOm8-hp?usp=sharing

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by rbharath** [comment]
@riya-singh28 @ARY2260 Please review

**Comment by riya-singh28** [comment]
@Harindhar10 @rbharath 
I went through the code given to replicate the issue, and here is what I think can be an issue. The infograph and infographstar models have different components, and the component 'unsup_encoder' is not present in the infograph. To use load_for_pretrained, the component layer architecture should be present in both the source model and the other model. While you don't see an issue when running the first code, the state_dict is not being updated there.

image

**Comment by Harindhar10** [comment]
Understood. I got the snippet from test_infograph_pretrain_overfit() function
